### ISSUE TITLE
Add flags for every takeable / buyable item

### ISSUE THREAD

**Issue by Dinopony** [Dinopony/randstalker#49]
Every item on ground / in shops in the game should have its own flag in order to be able to make it disappear once taken / bought.
These flags should replace the visited flags mechanism for maps (which is not used anymore in randomizer, since Oracle Stone has another function).

Technically, it's not so hard to setup:

1. Remove all visited flags for maps (set them all to 0xFF1000:0)
2. In `RandomizerWorld`, add a table for matching `ItemSource`s with `Flag`s
3. For every ground & shop `ItemSource` in this table, add the `Flag` as a clear flag to remove the entity when flag is set
4. Add a mechanism to set the flag in-game:
   - there must be a way to disable this mechanism for non-precious items (make an editable list of "infinite" items?), either by removing the clear flag or not setting the flag at all
   - it must set the flag when getting a "precious" ground item (be careful of code edits already done for armor upgrades)
   - it must set the flag when getting a "precious" shop item

Once the technical part is done, we can rework the pool of valid items in shops, and also change the prices of a few items. These items will benefit from these changes:
- **Life Stock** can be made precious and way cheaper (100g instead of 250g?) since we cannot take them several times
- **Pawn Ticket** can be made precious and slightly cheaper again (disabling potential abuses)
- **Kazalt Jewel** can be made precious, avoiding any kind of cheesing 6+ jewels obtention

It will also probably make sense to allow having duplicates in shops, as long as they are precious items. This will need to be considered on the randomization side of things.

**Comment by Dinopony** [comment]
Done in b24efa5
